src/NServiceBus.AcceptanceTests/Reliability/Outbox/When_a_duplicate_message_arrives.cs

```diff
@@ -0,0 +1,125 @@
+﻿namespace NServiceBus.AcceptanceTests.Reliability.Outbox
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.AcceptanceTests.ScenarioDescriptors;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+
+    public class When_a_duplicate_message_arrives : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task It_does_not_dispatch_messages_aready_dispatched()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<OutboxEndpoint>(b => b.When(async session =>
+                {
+                    var duplicateMessageId = Guid.NewGuid().ToString();
+
+                    var options = new SendOptions();
+
+                    options.SetMessageId(duplicateMessageId);
+                    options.RouteToThisEndpoint();
+
+                    await session.Send(new PlaceOrder(), options);
+                    await session.Send(new PlaceOrder(), options);
+                    await session.SendLocal(new PlaceOrder
+                    {
+                        Terminator = true
+                    });
+                }))
+                .WithEndpoint<DownstreamEndpoint>()
+                .Done(c => c.Done)
+                .Repeat(r => r.For<AllOutboxCapableStorages>())
+                .Should(context => Assert.AreEqual(2, context.MessagesReceivedByDownstreamEndpoint))
+                .Run();
+        }
+
+        public class Context : ScenarioContext
+        {
+            public int MessagesReceivedByDownstreamEndpoint { get; set; }
+            public bool Done { get; set; }
+        }
+
+        public class DownstreamEndpoint : EndpointConfigurationBuilder
+        {
+            public DownstreamEndpoint()
+            {
+                EndpointSetup<DefaultServer>(b =>
+                {
+                    b.LimitMessageProcessingConcurrencyTo(1);
+                    b.GetSettings().Set("DisableOutboxTransportCheck", true);
+                    b.EnableOutbox();
```

Comment:
wouldn't it make more sense to disable the outbox on the `DownStreamEndpoint` and verify that 3 messages arrived instead?